sheepdog/utils/transforms/graph_to_doc.py

```diff
@@ -458,12 +458,21 @@ def entity_to_template_json(links, schema, exclude_id):
         if not is_property_hidden(key, schema, exclude_id)
     }
     for key in properties:
+        if 'required' in schema and key in schema['required']:
+            marked_key = '*' + key
+        else:
+            marked_key = key
         if key in links:
-            keys[key] = links[key]
+            keys[marked_key] = links[key]
         elif key == 'type':
-            keys[key] = schema['id']
+            keys[marked_key] = schema['id']
         else:
-            keys[key] = None
+            keys[marked_key] = None
+    # users need to submit the 'urls' field, but 'urls' is not in the schema
+    # (since it is a property of records in indexd, and is not in the dict).
+    # So we are adding it to the templates here.
+    if schema['category'] == 'data_file':
```

Comment:
`urls` won't be required, since a user could still submit metadata before having any data files in storage